File: program_data/github_java_sort_function/6/41.java

public class RadixSort{
    public int getMax(int[] arr){
        int max = arr[0];
        for(int i = 0; i < arr.length; i++){
            if(arr[i] > max){
                max = arr[i];
            }
        }
        return max;
    }
    public void countSort(int[] arr, int n, int exp){
        int[] count = new int[n];
        int[] output = new int[arr.length];
        for(int i = 0; i < arr.length; i++){
            count[(arr[i]/exp)%n] ++;
        }
        for(int i = 1; i < n; i++){
            count[i] += count[i-1];
        }
        for(int i = arr.length - 1; i >= 0; i--){
            output[count[(arr[i]/exp)%n] - 1] = arr[i];
            count[(arr[i]/exp)%n] --;
        }
        for(int i = 0; i < arr.length; i++){
            arr[i] = output[i];
        }
    }
    public void radixSort(int[] arr){
        int max = getMax(arr);
        for(int exp = 1; max/exp > 0; exp *= arr.length){
            countSort(arr, arr.length, exp);
        }
    }

}

File: program_data/PKU_raw/75/41.c
int main()
{
	int a[1000]={0},i=0,b[1000][2]={0},n=0,j,max;
	char c;
	do
	{
		scanf("%d%c",&b[i][0],&c);
		n++;
		i++;
	}while(c!='\n');
	i=0;
	c=0;
	do
	{
		scanf("%d%c",&b[i][1],&c);
		i++;
	}while(c!='\n');
	for(j=0;j<1000;j++)
	{
		for(i=0;i<n;i++)
		{
			if(j>=b[i][0]&&j<b[i][1]) a[j]++;
		}
	}
	max=a[0];
	for(i=1;i<1000;i++) if(max<a[i]) max=a[i];
	printf("%d %d",n,max);
}